api/src/pool_queue.c

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2016, ARM Limited, All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "api/inc/export_table_exports.h"
+#include "api/inc/pool_queue_exports.h"
+
+int uvisor_pool_init(uvisor_pool_t * pool, void * array, size_t stride, size_t num, int blocking)
+{
+    return uvisor_export_table()->pool_queue.pool_init(pool, array, stride, num, blocking);
```

Comment:
@jaeden - consider storing `uvisor_export_table()->pool_queue` into a user mode pointer that is used directly and initialised during user mode init.